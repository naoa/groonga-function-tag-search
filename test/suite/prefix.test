register functions/tag_search

table_create Tags TABLE_PAT_KEY ShortText

table_create Entries TABLE_NO_KEY
column_create Entries title COLUMN_SCALAR ShortText
column_create Entries tag COLUMN_SCALAR Tags

load --table Entries
[
{"title": "Groonga and NoSQL", "tag": "No*SQL"},
{"title": "Mroonga and MySQL", "tag": "MySQL"},
{"title": "PGroonga and PostgreSQL", "tag": "PostgreSQL"}
]

table_create Terms TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram
column_create Terms title_index COLUMN_INDEX|WITH_POSITION Entries title

column_create Tags tag_index COLUMN_INDEX Entries tag

select Entries --filter 'title @ "roonga" && tag_search(tag, "No*SQ*", "MySQL", {"op": "OR"})' --output_columns '_id,_score,title,tag'
