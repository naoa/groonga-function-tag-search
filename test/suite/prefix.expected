register functions/tag_search
[[0,0.0,0.0],true]
table_create Tags TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
table_create Entries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Entries title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Entries tag COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
load --table Entries
[
{"title": "Groonga and NoSQL", "tag": "No*SQL"},
{"title": "Mroonga and MySQL", "tag": "MySQL"},
{"title": "PGroonga and PostgreSQL", "tag": "PostgreSQL"}
]
[[0,0.0,0.0],3]
table_create Terms TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram
[[0,0.0,0.0],true]
column_create Terms title_index COLUMN_INDEX|WITH_POSITION Entries title
[[0,0.0,0.0],true]
column_create Tags tag_index COLUMN_INDEX Entries tag
[[0,0.0,0.0],true]
select Entries --filter 'title @ "roonga" && tag_search(tag, "No*SQ*", "MySQL", {"op": "OR"})' --output_columns '_id,_score,title,tag'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_score",
          "Int32"
        ],
        [
          "title",
          "ShortText"
        ],
        [
          "tag",
          "Tags"
        ]
      ],
      [
        1,
        2,
        "Groonga and NoSQL",
        "No*SQL"
      ],
      [
        2,
        2,
        "Mroonga and MySQL",
        "MySQL"
      ]
    ]
  ]
]
